src/EventStore.Client/EventStoreGrpcClient.Read.cs

```diff
@@ -181,19 +185,27 @@ public partial class EventStoreClient {
 
 			await foreach (var e in call.ResponseStream
 				.ReadAllAsync(cancellationToken)
-				.Select(ConvertToResolvedEvent)
+				.Select(ConvertToItem)
 				.WithCancellation(cancellationToken)
 				.ConfigureAwait(false)) {
 				yield return e;
 			}
 
-			ResolvedEvent ConvertToResolvedEvent(ReadResp response) =>
+			(SubscriptionConfirmation, ResolvedEvent) ConvertToItem(ReadResp response) => response.ContentCase switch {
```

Comment:
We might want to do some bench marks around this. I believe it is just destructuring anonymous tuples that has performance implications, so I don't think we are affected by it here, but we should check in the RC time frame